File: scala/sublist/src/main/scala/Sublist.scala
object Sublist extends Enumeration {
  val Equal, Unequal, Sublist, Superlist = Value

  def sublist(a: List[Int], b: List[Int]) = if (a equals b) Equal
  else if (a containsSlice b) Superlist else if (b containsSlice a) Sublist else Unequal
}

File: php/sublist/Sublist.php
<?php

final class Sublist
{
    public function compare(array $listOne, array $listTwo): string {
        $joined_list_1 = implode(',', $listOne) . ',';
        $joined_list_2 = implode(',', $listTwo) . ',';
        
        $oneInTwo = str_contains($joined_list_2, $joined_list_1);
        $twoInOne = str_contains($joined_list_1, $joined_list_2);
        
        return $oneInTwo && $twoInOne ? 'EQUAL' : ($oneInTwo ? 'SUBLIST' : ($twoInOne ? 'SUPERLIST' : 'UNEQUAL'));
    }
}
